File: Lista V (C++) - Aula 22/C1_02/Departamento.cpp
#include <iostream>
#include "Departamento.h"

//Construtores
Departamento::Departamento(const string& nome) {
	this->setNome(nome);
}

//GETs & SETs

/*Sobre o retorno:
  string Departamento::getNome() // faço cópia no retorno
  string& Departamento::getNome() // PROIBIDO!!!!
  const string& // é eficiente (não faz cópia) e não deixa modificar do lado de fora do objeto
*/
const string& Departamento::getNome() const {
	return nome;
}
void Departamento::setNome(const string& nome){
	this->nome = nome;
}

//Métodos

Funcionario* Departamento::getFuncionario(int idx) const {
	return funcionarios[idx];
}

void Departamento::adicionarFuncionario(Funcionario* funcionario) {
	funcionarios.push_back(funcionario);
}

void Departamento::removerFuncionario(int idx) {
	funcionarios.erase(begin(funcionarios) + idx);
}

void Departamento::darAumento(const double percentual) {
	for(auto& f : funcionarios) //for para todo funcionario f do vetor 
		f->darAumento(percentual);
}

double Departamento::calcularCusto() const {
	double soma = 0;
	for (auto& f : funcionarios)
		soma += f->getSalario();
	return soma;
}

Departamento::~Departamento() {
	for (auto f : funcionarios)
		delete f;
}

File: Lista II (JAVA) - Aula 4/J1_08/Departamento.java
public class Departamento {
    String nome;
    Funcionario[] funcionarios;

    Departamento() {}

    Departamento(int num){
        this.funcionarios = new Funcionario[num];
    }

    public void setNome (String nome) {
        this.nome = nome;
    }

    String getNome(){
        return this.nome;
    }

    public void setFuncionario (int j, Funcionario funcionario) {
        this.funcionarios[j] = funcionario;
    }

    Funcionario getFuncionario(int i){
        return this.funcionarios[i];
    }
}
